### PR TITLE
RCAL-185 Update "dark" step to use STCAL

### LINKED ISSUE #271: Update "dark" step to use STCAL
_Issue [RCAL-185](https://jira.stsci.edu/browse/RCAL-185) was created on JIRA by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe):_

Update the step file to use RDM objects
Update the process file to use RDM objects
Update tests to use RDM objects

### PR DESCRIPTION
**Pull request by PaulHuwe**
… step.

GitHub issue, Closes #271

Resolves [RCAL-185](https://jira.stsci.edu/browse/RCAL-185)

**Description**

This PR updated dark current step to use stcal. Created tests for the updated step.

Regression test updates will be in RCAL-285.

Checklist
- [x] Tests

- [ ] Documentation

- [x] Change log

- [x] Milestone

- [x] Label(s)

### CODE REVIEW

**Review by ddavis-stsci — CHANGES_REQUESTED** [review]
In addition the regression tests are failing for me locally. 
=========================== short test summary info ============================
FAILED ../../../romancal/romancal/regtest/test_dark_current.py::test_dark_current_subtraction_step
FAILED ../../../romancal/romancal/regtest/test_dark_current.py::test_dark_current_outfile_step
FAILED ../../../romancal/romancal/regtest/test_dark_current.py::test_dark_current_outfile_suffix
=========== 3 failed, 13 deselected, 1 xfailed in 700.30s (0:11:40) ============
Are there artifactory updates that need to be done as well?

File: romancal/dark_current/tests/test_dark.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    # Set test size
+    shape = (2, 20, 20)
+
+    # Create test ramp and dark models
+    ramp_model, darkref_model = create_ramp_and_dark(shape, instrument, exptype)
+
+    # Add values for CRDS mismatch
+    ramp_model.meta.observation.start_time = Time('1922-01-01T11:11:11.110')
+    ramp_model.meta.observation.end_time = Time('1922-01-01T11:33:11.110')
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/dark_current/tests/test_dark.py:142]: The regression tests are addressed in RCAL-285.

[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/dark_current/tests/test_dark.py:142]: Year updated to 1999

File: romancal/dark_current/tests/test_dark.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+def create_ramp_and_dark(shape, instrument, exptype):
+    """Helper function to create test ramp and dark models"""
+
+    # Create test ramp model
+    ramp = testutil.mk_ramp(shape)
+    ramp.meta.instrument.name = instrument
+    ramp.meta.instrument.detector = 'WFI01'
+    ramp.meta.instrument.optical_element = 'F158'
+    ramp.meta.exposure.type = exptype
+    ramp.data = np.ones(shape, dtype=np.uint16)
```
[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/dark_current/tests/test_dark.py]: Changed... this was a holdover from a jwst test.

[the remark this answers is not in the document]
Comment (PaulHuwe) [reply romancal/dark_current/tests/test_dark.py]: @ddavis-stsci 
There is a line from the step, that I preserved, which reads:

line 46:
input_model.data = input_model.data.astype(np.float32)[np.newaxis, :]

Should I remove that casting, or keep it for safety?

[the remark this answers is not in the document]
Comment (ddavis-stsci) [reply romancal/dark_current/tests/test_dark.py]: I'd keep it for safety. My guess is that someone/something managed to trigger it.

**Review by ddavis-stsci — APPROVED** [review]
Looks good

**Pull request merged**